pkgs/applications/audio/vcv-rack/default.nix

```diff
@@ -56,46 +112,96 @@ with lib; stdenv.mkDerivation rec {
     # above), we do it here manually
     mkdir -p dep/include
 
-    cp -r ${pfft-source} dep/jpommier-pffft-source
+    cp -r ${pffft-source}/* dep/pffft
+    cp -r ${fuzzysearchdatabase-source}/* dep/fuzzysearchdatabase
     cp -r ${nanovg-source}/* dep/nanovg
     cp -r ${nanosvg-source}/* dep/nanosvg
     cp -r ${osdialog-source}/* dep/osdialog
     cp -r ${oui-blendish-source}/* dep/oui-blendish
 
-    cp dep/jpommier-pffft-source/*.h dep/include
+    cp dep/pffft/*.h dep/include
+    cp dep/fuzzysearchdatabase/src/*.hpp dep/include
     cp dep/nanosvg/**/*.h dep/include
     cp dep/nanovg/src/*.h dep/include
     cp dep/osdialog/*.h dep/include
     cp dep/oui-blendish/*.h dep/include
 
-    substituteInPlace include/audio.hpp --replace "<RtAudio.h>" "<rtaudio/RtAudio.h>"
-    substituteInPlace compile.mk --replace "-march=nocona" ""
+    # Build and dist the Fundamental plugins
+    cp -r ${fundamental-source} plugins/Fundamental/
+    chmod -R +rw plugins/Fundamental # will be used as build dir
+    substituteInPlace plugin.mk --replace ":= all" ":= dist"
+
+    # Fix reference to zenity
+    substituteInPlace dep/osdialog/osdialog_zenity.c \
+      --replace 'zenityBin[] = "zenity"' 'zenityBin[] = "${gnome.zenity}/bin/zenity"'
   '';
 
   enableParallelBuilding = true;
 
-  nativeBuildInputs = [ makeWrapper pkg-config ];
-  buildInputs = [ alsa-lib curl glew glfw gtk2-x11 jansson libjack2 libsamplerate libzip rtaudio rtmidi speex ];
+  nativeBuildInputs = [
+    copyDesktopItems
+    imagemagick
+    jq
+    libicns
+    makeWrapper
+    pkg-config
+    wrapGAppsHook
+  ];
+  buildInputs = [
+    alsa-lib
+    curl
+    ghc_filesystem
+    glew
+    glfw
+    gnome.zenity
+    gtk3-x11
+    jansson
+    libarchive
+    libjack2
+    libpulseaudio
+    libsamplerate
+    rtaudio
+    rtmidi
+    speex
+    zstd
+  ];
 
-  buildFlags = [ "Rack" ];
+  makeFlags = [ "all" "plugins" ];
 
   installPhase = ''
+    runHook preInstall
+
     install -D -m755 -t $out/bin Rack
+    install -D -m755 -t $out/lib libRack.so
 
     mkdir -p $out/share/vcv-rack
-    cp -r res Core.json template.vcv LICENSE* cacert.pem $out/share/vcv-rack
+    cp -r res cacert.pem Core.json template.vcv LICENSE-GPLv3.txt $out/share/vcv-rack
+    cp -r plugins/Fundamental/dist/Fundamental-*.vcvplugin $out/share/vcv-rack/Fundamental.vcvplugin
 
     # Override the default global resource file directory
     wrapProgram $out/bin/Rack --add-flags "-s $out/share/vcv-rack"
```

Comment:
You need to prevent double wrapping. It is currently wrapped 3 times.

Reply:
I'm going to be honest here; I'm not very familiar with how this wrapping works and carried this line over from previous maintainers' work. What are the other two places where the wrapping occurs? If I know what they are I'll try to consolidate them down to one.

Reply:
I've been reading a little bit about this topic and it looks like i can use something like `makeWrapperArgs+=("''${gappsWrapperArgs[@]}")` combined with `dontWrapGApps = true;`, then also incoroporate the arguments from this `wrapProgram` directive. I'll try this but if I'm not on the right track, let me know.

Reply:
I appreciate your honesty :). In general, one should not perform a wrap in an `installPhase`, but in the `fixupPhase`. Here's why: Here, because there's `wrapGAppsHook` in the `nativeBuildInputs`, what happens is that you wrap `Rack` in the `installPhase`, and then `wrapGAppsHook` wraps both the wrapped binary and the wrapper with the gapps environment variables. So I'd start testing with something like this:

```
diff --git i/pkgs/applications/audio/vcv-rack/default.nix w/pkgs/applications/audio/vcv-rack/default.nix
index e197bd8e39f..05ce2fb7290 100644
--- i/pkgs/applications/audio/vcv-rack/default.nix
+++ w/pkgs/applications/audio/vcv-rack/default.nix
@@ -28,7 +28,8 @@
 , speex
 , stdenv
 , wrapGAppsHook
-, zstd }:
+, zstd
+}:
 
 let
   # The package repo vendors some of the package dependencies as submodules.
@@ -178,9 +179,6 @@ stdenv.mkDerivation rec {
     cp -r res cacert.pem Core.json template.vcv LICENSE-GPLv3.txt $out/share/vcv-rack
     cp -r plugins/Fundamental/dist/Fundamental-*.vcvplugin $out/share/vcv-rack/Fundamental.vcvplugin
 
-    # Override the default global resource file directory
-    wrapProgram $out/bin/Rack --add-flags "-s $out/share/vcv-rack"
-
     # Extract pngs from the Apple icon image and create
     # the missing ones from the 1024x1024 image.
     icns2png --extract icon.icns
@@ -195,6 +193,11 @@ stdenv.mkDerivation rec {
     runHook postInstall
   '';
 
+  preFixup = ''
+    # Override the default global resource file directory
+    makeWrapperArgs=('--add-flags -s $out/share/vcv-rack')
+  '';
+
   meta = with lib; {
     description = "Open-source virtual modular synthesizer";
     homepage = "https://vcvrack.com/";
```

Applied on top of the current changes.

Reply:
It might take you several full compilation attempts to see that it's wrapped properly - you may encounter quoting issues there in the `makeWrapperArgs` variable, or you may need to use `gappsWrapperArgs` like so:

```nix
gappsWrapperArgs+=('--add-flags')
gappsWrapperArgs+=('-s')
gappsWrapperArgs+=("$out/share/vcv-rack")
```

Reply:
> I've been reading a little bit about this topic and it looks like i can use something like `makeWrapperArgs+=("''${gappsWrapperArgs[@]}")` combined with `dontWrapGApps = true;`, then also incoroporate the arguments from this `wrapProgram` directive. I'll try this but if I'm not on the right track, let me know.

That's also an option, perhaps better then my proposal.

Reply:
I amended my commit. I had good luck building the derivation using this technique adapted from the bitwig Nix derivations:

```nix
  dontWrapGApps = true;
  postFixup = ''
    # Wrap gApp and override the default global resource file directory
    wrapProgram $out/bin/Rack \
        "''${gappsWrapperArgs[@]}" \
        --add-flags "-s $out/share/vcv-rack"
  '';
```

Reply:
Uh, I had forgotten to actually add my changes to the commit. _Now_ I've pushed them. :sweat_smile: